samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -217,25 +216,44 @@ class OffsetManager(
   }
 
   /**
-   * Checkpoint all offsets for a given TaskName using the CheckpointManager.
-   */
-  def checkpoint(taskName: TaskName) {
+    * Gets a snapshot of all the current offsets for the specified task. This is useful to
+    * ensure there are no concurrent updates to the offsets between when this method is
+    * invoked and the corresponding call to [[OffsetManager.writeCheckpoint()]]
+    */
+  def createCheckpoint(taskName: TaskName): Checkpoint = {
```

Comment:
So I originally called it getCheckpoint() but found myself asking "does it get the value of the latest written checkpoint, or does it create a new one?"

In the spirit of compromise, how about getNewCheckpoint() or buildCheckpoint()?